homeassistant/components/sensor/tado_v1.py

```diff
@@ -0,0 +1,197 @@
+                'link', 'heating', 'tado mode', 'overlay']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    #  get the PyTado object from the hub component
+    tado = hass.data[CONST_TADO_DATA]
+
+    try:
+        zones = tado.get_zones()
+    except RuntimeError:
+        _LOGGER.error("Unable to get zone info from mytado")
+        return False
+
+    sensor_items = []
+    for zone in zones:
+        if zone['type'] == 'HEATING':
+            for variable in SENSOR_TYPES:
+                sensor_items.append(create_zone_sensor(
+                    tado, zone, zone['name'], zone['id'],
+                    variable))
+
+    me_data = tado.get_me()
+    sensor_items.append(create_device_sensor(
+        tado, me_data, me_data['homes'][0]['name'],
+        me_data['homes'][0]['id'], "tado bridge status"))
+
+    tado.update()
+
+    if len(sensor_items) > 0:
+        add_devices(sensor_items)
+        return True
+    else:
+        return False
+
+
+def create_zone_sensor(tado, zone, name, zone_id, variable):
+    """Create a zone sensor."""
+    data_id = 'zone {} {}'.format(name, zone_id)
+
+    tado.add_sensor(data_id, {
+        "zone": zone,
+        "name": name,
+        "id": zone_id,
+        "data_id": data_id
+    })
+
+    return TadoSensor(tado, name, zone_id, variable, data_id)
+
+
+def create_device_sensor(tado, device, name, device_id, variable):
+    """Create a device sensor."""
+    data_id = 'device {} {}'.format(name, device_id)
+
+    tado.add_sensor(data_id, {
+        "device": device,
+        "name": name,
+        "id": device_id,
+        "data_id": data_id
+    })
+
+    return TadoSensor(tado, name, device_id, variable, data_id)
+
+
+class TadoSensor(Entity):
+    """Representation of a tado Sensor."""
+
+    def __init__(self, store, zone_name, zone_id, zone_variable, data_id):
+        """Initialization of TadoSensor class."""
+        self._store = store
+
+        self.zone_name = zone_name
+        self.zone_id = zone_id
+        self.zone_variable = zone_variable
+
+        self._unique_id = '{} {}'.format(zone_variable, zone_id)
+        self._data_id = data_id
+
+        self._state = None
+        self._state_attributes = None
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.zone_name, self.zone_variable)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state_attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.zone_variable == "temperature":
+            return TEMP_CELSIUS
```

Comment:
Change this to `return self.hass.config.units.temperature_unit` (see comment below)